skops/card/_card_alternative.py

```diff
@@ -0,0 +1,361 @@
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass, field
+from pathlib import Path
+from reprlib import Repr
+from typing import Any, Iterator, Protocol
+
+from huggingface_hub import CardData
+from sklearn.utils import estimator_html_repr
+from tabulate import tabulate  # type: ignore
+
+from skops.card._model_card import PlotSection, TableSection
+
+aRepr = Repr()
+aRepr.maxother = 79
+aRepr.maxstring = 79
+
+
+def split_subsection_names(key: str) -> list[str]:
+    return key.split("/")
+
+
+def _clean_table(table: str) -> str:
+    # replace line breaks "\n" with html tag <br />, however, leave end-of-line
+    # line breaks (eol_lb) intact
+    eol_lb = "|\n"
+    placeholder = "$%!?"  # arbitrary sting that never appears naturally
+    table = (
+        table.replace(eol_lb, placeholder)
+        .replace("\n", "<br />")
+        .replace(placeholder, eol_lb)
+    )
+    return table
+
+
+@dataclass
+class Section:
+    title: str
+    content: Formattable | str | None = None
+    subsections: dict[str, Section] = field(default_factory=dict)
+
+
+class Formattable(Protocol):
+    def format(self) -> str:
+        ...
+
+
+class Card:
+    @classmethod
+    def make_default(
```

Comment:
when is this called? In the constructor? We should have something which adds all the missing recommended sections by default.

Reply:
The idea is that if we want to have a model card with the defaults pre-filled as they currently are, we would call `card = Card.make_default(...)`. Of course, that would be bc breaking if `__init__` doesn't pre-fill. It's more intuitive IMO for `__init__` to create an empty card and for pre-filled ones to use class methods, but we can also change that.